### PR TITLE
-- 616: adding support for passing arbitrary arguments to the java swarm c…

### PR DESCRIPTION
**Pull request by madelaney**
This allows the end-user to give an arbitrary list of additional arguments to the swarm client.

### CODE REVIEW

File: manifests/slave.pp [hunk 1]
```diff
@@ -170,6 +176,16 @@
     }
   }
 
+  if $slave_args {
+    if is_array($slave_args) {
+      $_combined_slave_args = hiera_array('jenkins::slave::slave_args', $slave_args)
```
Comment (jhoblitt) [inline manifests/slave.pp]: Whoa -- I feel this is pretty surprising behavior.  What is the motivation for doing this?

Comment (madelaney) [inline manifests/slave.pp]: @jhoblitt fair question. The idea behind this is that there are flags, like `disableClientsUniqueId` and `deleteExistingClients`, that I want to send to the client. I thought of listing these out explicitly but felt this was a bit easier to manage going forward.

Comment (rtyler) [inline manifests/slave.pp]: I don't understand the need for a Hiera lookup though, you can already inject class parameters by defining:

```yaml
jenkins::slave::slave_args: "bippidyboppidy-boop"
```

Comment (madelaney) [inline manifests/slave.pp]: @rtyler , I did that so we can take advantage of the deep merge behavior.

Comment (rtyler) [inline manifests/slave.pp]: @madelaney I figured that might be the case, but I think that makes this behavior even more confusing. Hiera already has configuration (which I *think* applies to injected class parameters) for its [merge behaviors](https://docs.puppet.com/hiera/3.2/configuring.html#mergebehavior) (also [array merge handling](https://docs.puppet.com/hiera/3.2/lookup_types.html#array-merge)).

There's also [this documentation](https://docs.puppet.com/puppet/latest/reference/lookup_quick.html#specifying-merge-behavior) which might be relevant? The myriad of merge-related documents annoys and confuses me :person_frowning: 


Anyways, I generally find accepting class parameters *plus* doing a `hiera_array` lookup of the same key to be weird.

Comment (madelaney) [inline manifests/slave.pp]: @rtyler, fair enough. The docs can be very confusing. If I change line 181-182 to read more like:
```
$_real_slave_args = join($slave_args, ' ')
```
Would you be okay with that?

**Pull request closed without merging**